web/server.go

```diff
@@ -35,10 +36,14 @@ func NewMux(config Config, context command.ExecutionContext, hook Hook) *http.Se
 	httpMux.Handle("/token", tokenHandler{
 		context: context,
 	})
-	if config.JSONIngestion {
-		httpMux.Handle("/ingest", ingestHandler{
-			metricMetadataAPI: context.MetricMetadataAPI,
-		})
+	if updateAPI, ok := context.MetricMetadataAPI.(metadata.MetricUpdateAPI); ok && config.HTTPIngestion {
```

Comment:
It might even be okay to go `Fatal`.

Right now one issue is that it's incompatible with the Caching API, but I can fix that with some interfaces.